public/js/reducers/sources.js

```diff
@@ -21,7 +21,7 @@ export type SourcesState = {
 
 const State = makeRecord(({
   sources: I.Map(),
-  selectedSource: undefined,
+  selectedLocation: undefined,
```

Comment:
This is a nice improvement. There are many things I like here:
+ we're including the selected line
+ using a known type (location)
+ no longer copying the full source object.